avatars/lib/saferound_test.py

```diff
@@ -1,116 +1,114 @@
+    in_odict: OrderedDictType[str, float]
+    in_tuple: Tuple[float, float, float]
+    neg_in_list: list[float]
+    huge_in_list: list[float]
+
 
 @pytest.fixture
-def setup():
+def setup() -> Setup:
     in_list = [4.0001, 3.2345, 3.2321, 6.4523, 5.3453, 7.3422]
-    in_dict = {'foo': 60.19012964572332,
-                    'bar': 15.428802458406679,
-                    'baz': 24.381067895870007}
+    in_dict = {
+        "foo": 60.19012964572332,
+        "bar": 15.428802458406679,
+        "baz": 24.381067895870007,
+    }
     in_odict = OrderedDict(in_dict)
     in_tuple = (60.1901296, 15.42880, 24.38106789)
 
-    neg_in_list = [x * -1. for x in in_list]
-    huge_in_list = [x * 100000. for x in in_list]
+    neg_in_list = [x * -1.0 for x in in_list]
+    huge_in_list = [x * 100000.0 for x in in_list]
     setup = Setup(in_list, in_dict, in_odict, in_tuple, neg_in_list, huge_in_list)
     return setup
 
-def test_basic_difference(setup):
+
+def test_basic_difference(setup: Setup) -> None:
     out = [4.0, 3.24, 3.23, 6.45, 5.35, 7.34]
     assert saferound(setup.in_list, 2) == out
 
-def test_basic_largest(setup):
+
+def test_basic_largest(setup: Setup) -> None:
     out = [4.0, 3.23, 3.23, 6.45, 5.35, 7.35]
-    assert saferound(setup.in_list, 2, LARGEST) == out
+    assert saferound(setup.in_list, 2, RoundingStrategy.LARGEST) == out
+
 
-def test_basic_smallest(setup):
+def test_basic_smallest(setup: Setup) -> None:
     out = [4.0, 3.24, 3.23, 6.45, 5.35, 7.34]
-    assert saferound(setup.in_list, 2, SMALLEST) == out
+    assert saferound(setup.in_list, 2, RoundingStrategy.SMALLEST) == out
+
 
-def test_dict(setup):
-    out = {'foo': 60.0,
-            'bar': 16.0,
-            'baz': 24.0}
+def test_dict(setup: Setup) -> None:
+    out = {"foo": 60.0, "bar": 16.0, "baz": 24.0}
     assert saferound(setup.in_dict, 0) == out
 
-def test_odict(setup):
-    out = OrderedDict({'foo': 60.0,
-                        'bar': 16.0,
-                        'baz': 24.0})
+
+def test_odict(setup: Setup) -> None:
+    out = OrderedDict({"foo": 60.0, "bar": 16.0, "baz": 24.0})
     assert saferound(setup.in_odict, 0) == out
 
-def test_tuple(setup):
-    out = (60., 16., 24.)
+
+def test_tuple(setup: Setup) -> None:
+    out = (60.0, 16.0, 24.0)
     assert saferound(setup.in_tuple, 0) == out
 
-def test_error_bad_float(setup):
-    bad = ['a', 3.2345, 3.2321, 6.4523, 5.3453, 7.3422]
-    with pytest.raises(
-        AssertionError
-    ):
+
+def test_error_bad_float() -> None:
+    bad = ["a", 3.2345, 3.2321, 6.4523, 5.3453, 7.3422]
+    with pytest.raises(ValueError, match="all values in the iterable to be float"):
         saferound(bad, 2)
 
-def test_error_bad_places(setup):
-    with pytest.raises(
-        AssertionError
-    ):
-        saferound(setup.in_list, 2.5)
-
-def test_error_bad_strategy(setup):
-    with pytest.raises(
-        AssertionError
-    ):
-        saferound(setup.in_list, 2, strategy='foo')
```

Comment:
This test was testing validation is done at runtime (see my other comment). Why did you remove it ?